### ISSUE TITLE
1.6.0: documentation build fails

### ISSUE THREAD

**Issue by kloczek** [pywbem/pywbem#2983]
Looks like with latest sphinx 6.3.1 it is nt possible to build documentation
```console
+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v6.1.3
WARNING: Invalid configuration value found: 'language = None'. Update your configuration to a valid language code. Falling back to 'en' (English).
[autosummary] generating autosummary for: appendix.rst, changes.rst, client.rst, client/conversion.rst, client/exceptions.rst, client/logging.rst, client/objects.rst, client/operations.rst, client/proxy.rst, client/security.rst, ..., development.rst, genindex.rst, index.rst, indication.rst, intro.rst, mockwbemserver.rst, server.rst, subscription.rst, tutorial.rst, utilities.rst
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.python.org/2/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
building [mo]: targets for 0 po files that are out of date
writing output...
building [man]: all manpages
updating environment: [new config] 28 added, 0 changed, 0 removed
reading sources... [  3%] appendix
reading sources... [  7%] changes
reading sources... [ 10%] client
reading sources... [ 14%] client/conversion
reading sources... [ 17%] client/exceptions
reading sources... [ 21%] client/logging
reading sources... [ 25%] client/objects
reading sources... [ 28%] client/operations
reading sources... [ 32%] client/proxy
reading sources... [ 35%] client/security
reading sources... [ 39%] client/statistics
reading sources... [ 42%] client/status
reading sources... [ 46%] client/types
reading sources... [ 50%] client/units
reading sources... [ 53%] client/valuemappings
reading sources... [ 57%] client/warnings
reading sources... [ 60%] compiler
reading sources... [ 64%] concepts
reading sources... [ 67%] development
reading sources... [ 71%] genindex
reading sources... [ 75%] index
reading sources... [ 78%] indication
reading sources... [ 82%] intro
reading sources... [ 85%] mockwbemserver
reading sources... [ 89%] server
reading sources... [ 92%] subscription
reading sources... [ 96%] tutorial
/home/tkloczko/rpmbuild/BUILD/pywbem-1.6.0/pywbem_mock/_instancewriteprovider.py:docstring of pywbem_mock._instancewriteprovider.InstanceWriteProvider:1: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/tkloczko/rpmbuild/BUILD/pywbem-1.6.0/pywbem_mock/_instancewriteprovider.py:docstring of pywbem_mock._instancewriteprovider.InstanceWriteProvider:3: WARNING: Definition list ends without a blank line; unexpected unindent.

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 299, in next_line
    self.line = self.input_lines[self.line_offset]
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 1136, in __getitem__
    return self.data[i]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 226, in run
    self.next_line()
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 302, in next_line
    raise EOFError
EOFError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, args.filenames)
  File "/usr/lib/python3.8/site-packages/sphinx/application.py", line 347, in build
    self.builder.build_update()
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 308, in build_update
    self.build(['__all__'], to_build)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 327, in build
    updated_docnames = set(self.read())
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 434, in read
    self._read_serial(docnames)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 455, in _read_serial
    self.read_doc(docname)
  File "/usr/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 511, in read_doc
    publisher.publish()
  File "/usr/lib/python3.8/site-packages/docutils/core.py", line 224, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/usr/lib/python3.8/site-packages/sphinx/io.py", line 105, in read
    self.parse()
  File "/usr/lib/python3.8/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/usr/lib/python3.8/site-packages/sphinx/parsers.py", line 80, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 169, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2785, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2785, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 325, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 391, in new_subsection
    newabsoffset = self.nested_parse(
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 233, in run
    context, next_state, result = self.check_line(
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 445, in check_line
    return method(match, context, next_state)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 1667, in simple_table_top
    return self.table_top(match, context, next_state,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 1674, in table_top
    nodelist, blank_finish = self.table(isolate_function, parser_class)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 1692, in table
    table = self.build_table(tabledata, tableline)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 1816, in build_table
    tbody += self.build_table_row(row, tableline)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 1833, in build_table_row
    self.nested_parse(cellblock, input_offset=tableline+offset,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 279, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    result = state.eof(context)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2725, in eof
    self.blank(None, context, None)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2716, in blank
    paragraph, literalnext = self.paragraph(
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 416, in paragraph
    textnodes, messages = self.inline_text(text, lineno)
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 425, in inline_text
    nodes, messages = self.inliner.parse(text, lineno,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 649, in parse
    before, inlines, remaining, sysmessages = method(self, match,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 792, in interpreted_or_phrase_ref
    nodelist, messages = self.interpreted(rawsource, escaped, role,
  File "/usr/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 889, in interpreted
    nodes, messages2 = role_fn(role, rawsource, text, lineno, self)
  File "/usr/lib/python3.8/site-packages/sphinx/ext/extlinks.py", line 103, in role
    title = caption % part
TypeError: not all arguments converted during string formatting

Exception occurred:
  File "/usr/lib/python3.8/site-packages/sphinx/ext/extlinks.py", line 103, in role
    title = caption % part
TypeError: not all arguments converted during string formatting
The full traceback has been saved in /tmp/sphinx-err-_hlpe09q.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

**Comment by andy-maier** [comment]
I was able to reproduce the error locally on MacOS with Python 3.9 and latest package levels, using `make builddocs` and these versions for the relevant packages:
```
Sphinx                        6.1.3
docutils                      0.18.1
```

On Python 3.7 with latest package levels, it succeeds, and the versions of the relevant packages were:
```
Sphinx                        5.3.0
docutils                      0.18.1
```

I need to investigate how to avoid that issue.

Also, there are several warnings in the success case, which should be addressed.

**Comment by andy-maier** [comment]
The fix was to add a `%s` to the caption field of the extlinks definition in conf.py:
```
extlinks = {
  'nbview': ('https://nbviewer.jupyter.org/github/pywbem/pywbem/blob/master/docs/notebooks/%s', '%s'),
  'nbdown': ('https://github.com/pywbem/pywbem/raw/master/docs/notebooks/%s', '%s')
}
```

sphinx-build in Sphinx 5.x even issued a warning to that regard:
```
WARNING: extlinks: Sphinx-6.0 will require a caption string to contain exactly one '%s' and all other '%' need to be escaped as '%%'.
WARNING: extlinks: Sphinx-6.0 will require a caption string to contain exactly one '%s' and all other '%' need to be escaped as '%%'.
```
but Sphinx 6 then causes the exception in docutils if that is not done. I'll open an issue against Sphinx for that.

**Comment by andy-maier** [comment]
Fixed in PR #2987 for 1.7.0, and will be rolled back into 1.6.1.

**Comment by andy-maier** [comment]
Version 1.6.1 with the fix has been released to Pypi.
